src/devices/machine/nscsi_cd.cpp

```diff
@@ -4,52 +4,59 @@
 #include "machine/nscsi_cd.h"
 #include "imagedev/chd_cd.h"
 
+#define VERBOSE 1
+#include "logmacro.h"
+
 DEFINE_DEVICE_TYPE(NSCSI_CDROM, nscsi_cdrom_device, "scsi_cdrom", "SCSI CD-ROM")
 
 nscsi_cdrom_device::nscsi_cdrom_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
-	nscsi_full_device(mconfig, NSCSI_CDROM, tag, owner, clock), cdrom(nullptr), bytes_per_sector(0), lba(0), cur_lba(0), blocks(0)
+	nscsi_full_device(mconfig, NSCSI_CDROM, tag, owner, clock), cdrom(nullptr), bytes_per_block(bytes_per_sector), lba(0), cur_sector(0)
 {
 }
 
 void nscsi_cdrom_device::device_start()
 {
 	nscsi_full_device::device_start();
-	bytes_per_sector = 2048;
-	save_item(NAME(block));
+	save_item(NAME(sector_buffer));
 	save_item(NAME(lba));
-	save_item(NAME(cur_lba));
-	save_item(NAME(blocks));
-	save_item(NAME(bytes_per_sector));
+	save_item(NAME(cur_sector));
+	save_item(NAME(bytes_per_block));
 }
 
 void nscsi_cdrom_device::device_reset()
 {
 	nscsi_full_device::device_reset();
 	cdrom = subdevice<cdrom_image_device>("image")->get_cdrom_file();
 	lba = 0;
-	blocks = 0;
-	cur_lba = -1;
+	cur_sector = -1;
 }
 
 MACHINE_CONFIG_START(nscsi_cdrom_device::device_add_mconfig)
 	MCFG_CDROM_ADD("image")
 	MCFG_CDROM_INTERFACE("cdrom")
 MACHINE_CONFIG_END
 
+void nscsi_cdrom_device::set_block_size(u32 block_size) 
+{
+	assert_always(bytes_per_sector % block_size == 0, "block size must be a factor of sector size");
+
+	bytes_per_block = block_size; 
+};
```

Comment:
Is it safe to call this after start?  If not it should assert that it isn't started.